profiler/internal/cmemprof/pprof.go

```diff
@@ -0,0 +1,91 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package cmemprof
+
+import (
+	"os"
+	"runtime"
+
+	"github.com/google/pprof/profile"
+)
+
+func (c *Profile) build() *profile.Profile {
+	// TODO: can we be sure that there won't be other allocation samples
+	// ongoing that write to the sample map? Right now it's called with c.mu
+	// held but it wouldn't be good if this (probably expensive) function
+	// was holding the same lock that recordAllocationSample also wants to
+	// hold
+	p := &profile.Profile{}
+	m := &profile.Mapping{
+		ID:   1,
+		File: os.Args[0], // XXX: Is there a better way to get the executable?
```

Comment:
I originally did the mapping like this because having the executable path in the mapping helped the pprof tool symbolize the samples back when I just had program counters. But I'm not sure if this is the right thing to do. Can we have the same "mapping" as the Go runtime allocation profile?